packages/forklift-console-plugin/src/components/page/StandardPageWithExpansion.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState } from 'react';
+
+import { RowProps, withTr } from '@kubev2v/common';
+import { Td, Th } from '@patternfly/react-table';
+
+import StandardPage, { StandardPageProps } from './StandardPage';
+
+export function withRowExpansion<T>({ CellMapper, isExpanded, toggleExpandFor }) {
+  const Enhanced = (props: RowProps<T>) => (
+    <>
+      {isExpanded && (
+        <Td
+          expand={{
+            rowIndex: props.resourceIndex,
+            isExpanded: isExpanded(props.resourceData),
+            onToggle: () => toggleExpandFor([props.resourceData]),
+          }}
+        />
+      )}
+      <CellMapper {...props} />
+    </>
+  );
+  Enhanced.displayName = `${CellMapper.displayName || 'CellMapper'}WithExpansion`;
+  return Enhanced;
+}
+
+export interface IdBasedExpansionProps<T> {
+  /**
+   * @returns string that can be used as an unique identifier
+   */
+  toId?: (item: T) => string;
+
+  /**
+   * onExpand is called when expand changes
+   */
+  onExpand?: (expandedIds: string[]) => void;
+
+  /**
+   * Expanded ids
+   */
+  expandedIds?: string[];
+}
+
+/**
+ * Adds ID based expansion to StandardPage component.
+ * Contract:
+ * 1. IDs provided with toId() function are unique and constant in time
+ */
+export function withIdBasedExpansion<T>({
```

Comment:
arrow function and please avoid nested components

Reply:
Hi @metalice this file is basically a copy of [StandardPageWithSelection](https://github.com/kubev2v/forklift-console-plugin/blob/main/packages/forklift-console-plugin/src/components/page/StandardPageWithSelection.tsx), but without selection, only expansion